app/javascript/components/ProfileForm.jsx

```diff
@@ -311,9 +312,69 @@ class ProfileForm extends React.Component {
     )
   }
 
+  renderUploadForm(index) {
+    let buttonProps = null;
+    let method = (this.state.mode === 'edit') ? 'PUT' : 'POST';
+    let path = (this.state.mode === "create") ? '/api/properties' : '/api/properties/' + this.state.id.toString();
+    if (this.state.mode === "edit") {
+      buttonProps = {
+        listType: 'picture',
+        defaultFileList: this.state.prevValues[index] === null ? [] : [{uid: this.state.prevValues[index].id, url: this.state.prevValues[index].image, name: this.state.prevValues[index].image.split("/").slice(-1).pop()}],
+        className: 'upload-list-inline'
+      }
+    }
+    return (
+      <div key={index}>
+        Property Form
+        <UploadButton {...buttonProps} />
+        <ActiveStorageProvider
+          endpoint={{
+            path: path,
+            model: 'Property',
+            attribute: 'form',
+            method: method,
+          }}
+          headers={{
+            'Content-Type': 'application/json'
+          }}
+          render={({ handleUpload, uploads, ready }) => (
```

Comment:
if you choose to build the render into a separate function, could use that here as well